dask/array/core.py

```diff
@@ -4220,7 +4230,10 @@ def asanyarray(a):
     elif isinstance(a, (list, tuple)) and any(isinstance(i, Array) for i in a):
         return stack(a)
     elif not isinstance(getattr(a, "shape", None), Iterable):
-        a = np.asanyarray(a)
+        if like is not None:
+            a = np.asarray(a, like=meta_from_array(like))
+        else:
+            a = np.asarray(a)
```

Comment:
We've switched from `asanyarray`  to `asarray` here. I'm wondering why this change was made

Reply:
This change was made because I was lazy to type and copied/pasted from the `asarray` definition and forgot to change it back. Thanks for catching this which is now fixed by https://github.com/dask/dask/pull/8054/commits/8323178bfe39beb19a5d4c7cfa2ab92e9eade763 .